src/NServiceBus.AcceptanceTesting/Support/CustomComponentBehavior.cs

```diff
@@ -0,0 +1,71 @@
+﻿namespace NServiceBus.AcceptanceTesting.Support
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public class CustomComponentBehavior<TContext> : IEndpointBehavior
+        where TContext : ScenarioContext
+    {
+        string name;
+        Func<TContext, CancellationToken, Task> onStart;
+        Func<TContext, Task> onStop;
+        Func<TContext, CancellationToken, Task> onStarted;
+
+        public CustomComponentBehavior(string name, Func<TContext, CancellationToken, Task> onStart, Func<TContext, Task> onStop, Func<TContext, CancellationToken, Task> onStarted)
+        {
+            Guard.AgainstNull(nameof(onStart), onStart);
+            Guard.AgainstNull(nameof(onStop), onStop);
+
+            this.name = name;
+            this.onStart = onStart;
+            this.onStop = onStop;
+            this.onStarted = onStarted ?? ((c, t) => Task.FromResult(0));
```

Comment:
maybe just assign null and do the null check before the invocation to safe some delegates?